docs/guide/database-insert-update.md

```diff
@@ -0,0 +1,159 @@
+  ```javascript
+  module.exports = {
+    extend: '@apostrophecms/piece-type',
+    // ...
+    methods (self) {
+      async addNewDog(initialInfo) {
+        // Generate a req object using the contributor role
+        const req = apos.task.getReq({ role: 'contributor' });
+        // Generate a blank dog data object.
+        let newDog = self.newInstance();
+        // Add our initial information to the object.
+        newDog = {
+          ...newDog,
+          ...initialInfo
+        };
+        // Insert the dog with the asynchronous `self.insert` method
+        const insertResult = await self.insert(req, newDog);
+
+        return insertResult;
+      }
+    }
+  };
+  ```
+  <template v-slot:caption>
+    modules/dog/index.js
+  </template>
+</AposCodeBlock>
+
+## Inserting pages
+
+As mentioned above, pages have a position within the page tree. The `insert` method for pages therefore requires additional arguments to place a new page in that hierarchy properly. Also [as described above](#updating-page-documents) we normally call the insert method directly from the `@apostrophecms/page` module, aliased in server-side code as `self.apos.page`.
+
+The arguments for the `self.apos.page.insert()` method are:
+
+| Argument | Required | Description |
+| -------- | -------- | ----------- |
+| `req` | TRUE | The associated request object. Using a provided `req` object is important for maintaining user role permissions. |
+|`_targetId` | TRUE | The `_id` of an existing page to use as a target when inserting the new page. `_home` and `_archive` are optional conveniences for the home page and [archived section](/reference/api/pages.md#moving-pages-to-the-archive), respectively. |
+|`_position` | TRUE | A numeric value will represent the zero-based child index under the `_targetId` page. `before`, `after`, `firstChild`, or `lastChild` values set the position within the page tree for the new page in relation to the target page (see `_targetId`). `before` and `after` insert the new page as a sibling of the target. `firstChild` and `lastChild` insert the new page as a child of the target. |
+| `page` | TRUE | The page document object. |
+| `options` | FALSE | An options object, primarily used for internal draft state management. |
+
+As with pieces, this process will normally begin by generating an empty page document with the `newInstance()` method, which should be done using a specific page type module.
+
+
+<AposCodeBlock>
+  ```javascript
+  let newPage = self.newInstance();
+
+  newPage = {
+    ...newPage,
+    ...initialInfo
+  };
+
+  await self.apos.page.insert(req, '_home', 'lastChild', newPage);
+  ```
+  <template v-slot:caption>
+    modules/special-page/index.js
+  </template>
+</AposCodeBlock>
+
+::: note
+Whenever updating or inserting a content document, do not attempt to directly set any property beginning with an underscore (`_`). Any property with a leading underscore, except for `_id`, **will not be saved to the database**.
```

Comment:
While true in A2, this is incorrect in A3. In A3, relationship fields are read/write at the apostrophe model API level.

That means that if product has a relationship to article, you can populate `._articles` with an array of articles and insert your product and it'll save that relationship. However note that it won't save the articles themselves, you have to do that separately first if the articles themselves are new or updated. This also means it's OK to use article objects with a strict projection in the array, since you're not modifying them, just the relationship to them (apostrophe only really cares about the `_id` property and, if there is a relationship schema, the `_relationship` property of each one).

Reply:
Ah, right. I'll remove this, then and leave the bit about _id.